packages/providers/src/coin-quantity.ts

```diff
@@ -27,7 +27,7 @@ export const coinQuantityfy = (coinQuantityLike: CoinQuantityLike): CoinQuantity
   const bnAmount = bn(amount);
   return {
     assetId: hexlify(assetId),
-    amount: bnAmount.toNumber() < 1 ? bn(1) : bnAmount,
+    amount: bnAmount.isNeg() || bnAmount.isZero() ? bn(1) : bnAmount,
```

Comment:
```suggestion
    amount: bnAmount.lt(1) ? bn(1) : bnAmount,
```

I can appreciate that based off [this comment](https://github.com/FuelLabs/fuels-ts/pull/1603#issuecomment-1881210134), `BN` ignores decimals so we can just check for negative / zero, however this is not obvious from the code. Wondering if using the built in `lt` function would be more succinct? Giving us both a `bn` comparison and the previous less than 1 use case.